storage/watchable_store_bench_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"math/rand"
+	"os"
+	"testing"
+)
+
+// Benchmarks on cancel function performance for unsynced watchers
+// in a WatchableStore. It creates 2*N watchers to populate unsynced
+// with a reasonably large number of watchers. And measures the time it
+// takes to cancel N watchers out of 2*N watchers. The performance is
+// expected to differ depending on the unsynced member implementation.
```

Comment:
@xiang90 That's a good point. I think it would depend on the real-world use case.
How would people use watcher and cancel?

Or should we just say it's an arbitrary number we are testing? @yichengq suggested
cancel 1/10 or half of watchers in previous comments. That was why we were testing
N cancel function calls out of 2*N watchers. Any thoughts? Please let me know.